docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,237 @@
+package examples
+
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.partiql.eval.Mode
+import org.partiql.eval.compiler.PartiQLCompiler
+import org.partiql.parser.PartiQLParser
+import org.partiql.planner.PartiQLPlanner
+import org.partiql.spi.Context
+import org.partiql.spi.SourceLocation
+import org.partiql.spi.catalog.Identifier
+import org.partiql.spi.catalog.Session
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.PErrorKind
+import org.partiql.spi.errors.PErrorListener
+import org.partiql.spi.errors.PRuntimeException
+import org.partiql.spi.errors.Severity
+import org.partiql.spi.types.PType
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+/**
+ * A unified error handling is introduced PartiQL system. [PRuntimeException] will be thrown from all components in PartiQL system
+ * to trap any errors or warnings. [PRuntimeException] warps a [PError] Java class which exposes information for users to write
+ * high quality error messages.
+ *
+ * Also [PErrorListener] is introduced to allow customized error handling. You can register a [PErrorListener] in context of
+ * major PartiQL components [PartiQLParser], [PartiQLPlanner] and [PartiQLCompiler]. It allows you to inspect each PError
+ * the component emitted and decide action on error. You may treat all warnings as errors, suppress all warnings, delay the error handling
+ * or throw custom exception with an error message.
+ *
+ * Note for the output when you choose to delay the error handling. The component will not stop and still return an output object
+ * In case the severity is [Severity.ERROR], the output may be incorrect or incomplete. You should discard the result.
+ * You may proceed with [Severity.WARNING], but be aware of the warning condition such as date accuracy loss.
+ */
+class ErrorHandlingChanges {
+    class CustomPErrorListener : PErrorListener {
+        val errorCollection: MutableList<PError> = mutableListOf()
+
+        override fun report(error: PError) {
+            when (error.severity.code()) {
+                // You may choose to customize the error message and then halt the flow by throwing the error, or you can record the error and then throw when it reaches maximum count
+                // In this case, we record only without throwing
+                Severity.ERROR -> errorCollection.add(error)
+                // You may choose to customize the error message, treat warning as an error and then halt the flow by throwing, or you can suppress all warnings.
```

Comment:
nit: let's add some line breaks for these comments. they are currently pretty long